### ISSUE TITLE
tabular environment is treated as a paragraph in the textlint AST

### ISSUE THREAD

**Issue by kn1cht** [textlint/textlint-plugin-latex2e#69]
## Problem

[textlint-rule-ja-no-mixed-period](https://github.com/textlint-ja/textlint-rule-ja-no-mixed-period/) points out "no period" errors in `tabular` environments.

![image](https://user-images.githubusercontent.com/16277200/106372507-be6c5a80-63b3-11eb-9c9b-26c02ebeafc5.png)

## Cause
```tex
\begin{table}[]
  \begin{tabular}{cc}
    This is & a table
  \end{tabular}
\end{table}
```

This LaTeX code is converted to an AST like below.
Since textlint-rule-ja-no-mixed-period treats the last `Str` node of a `Paragraph` node as the end of a sentence, converting a tabular environment to these nodes causes an unexpected error.

```js
{
  "type": "Document",
  "raw": "...",
  "range": [],
  "loc": {},
  "children": [
    {
      "loc": {},
      "range": [],
      "raw": "cc}\n        This is & a table",
      "type": "Paragraph",
      "children": [
        {
          "loc": {},
          "range": [],
          "raw": "cc",
          "type": "Str",
          "value": "cc"
        },
        {
          "loc": {},
          "range": [],
          "raw": "}\n        ",
          "type": "Html",
          "value": "}\n        "
        },
        {
          "loc": {},
          "range": [],
          "raw": "This",
          "type": "Str",
          "value": "This"
        },
        // ...
      ]
    }
  ]
}
```

## Solution

textlint-plugin-review seems to treat each item in a table as a `ListItem` node.
We can also wrap `Str` nodes into `ListItem` nodes.

https://github.com/orangain/textlint-plugin-review/pull/6

**Comment by tani** [comment]
Textlint AST provides `Table` node. Why not use it instead of `ListItem`?
I note you that the third party parser latex-utensils has no node for table unfortunately.

**Comment by kn1cht** [comment]
@tani 
`Table` nodes is surely exists in `markdown-to-ast ` but not in `@textlint/ast-node-types`.
Thus I couldn't use node types like `ASTnodetypes.Table'.

https://github.com/textlint/textlint/blob/master/docs/txtnode.md#type

**Comment by tani** [comment]
OMG, should we send request to ast-node-types?

**Comment by kn1cht** [comment]
Thanks for writing the issue.
I will wait answers and decide which node to use for tabluar environments.
